File: ldfs/src/main/kotlin/com/ldfs/main/FileReplicationController.java
package com.ldfs.main;

import com.ldfs.main.dto.request.ChunkReplicationRequest;
import jakarta.transaction.Transactional;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/files/replication")
public class FileReplicationController {

    @PostMapping("/chunkReplicationInit")
    @Transactional
    public void chunkReplicationInit(@RequestBody ChunkReplicationRequest chunkReplicationRequest) {

    }
}

File: ldfs/src/main/kotlin/com/ldfs/presentation/FileReplicationController.kt
package com.ldfs.presentation

import com.ldfs.presentation.dto.request.ChunkReplicationRequest
import jakarta.transaction.Transactional
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/files/replication")
class FileReplicationController {
    @PostMapping("/chunkReplicationInit")
    @Transactional
    fun chunkReplicationInit(
        @RequestBody chunkReplicationRequest: ChunkReplicationRequest?,
    ) {
    }
}
